src/watertap_contrib/seto/costing/solar/trough.py

```diff
@@ -0,0 +1,161 @@
+import pyomo.environ as pyo
+from watertap.costing.util import register_costing_parameter_block
+from watertap_contrib.seto.costing.util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+    make_variable_operating_cost_var,
+)
+
+
+def build_trough_cost_param_block(blk):
+
+    costing = blk.parent_block()
+
+    blk.base_storage_hours = pyo.Var(
+        initialize=6,
+        units=pyo.units.hour,
+        bounds=(6, 6),
+        doc="Hours of storage that capital cost per capacity is based on",
+    )
+
+    blk.cost_per_capacity_capital = pyo.Var(
+        initialize=560,
+        units=costing.base_currency / pyo.units.kW,
+        bounds=(0, None),
+        doc="Cost per kW (thermal) for the trough plant, assuming six hours storage",
+    )
+
+    blk.cost_per_storage_capital = pyo.Var(
+        initialize=62,
+        units=costing.base_currency / pyo.units.kWh,
+        bounds=(0, None),
+        doc="Cost per kWh (thermal) for the trough plant thermal storage",
+    )
+
+    blk.contingency_frac_direct_cost = pyo.Var(
+        initialize=0.07,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs for contingency",
+    )
+
+    blk.tax_frac_direct_cost = pyo.Var(
+        initialize=0.05,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs applicable for sales tax",
+    )
+
+    blk.sales_tax_rate = pyo.Var(
+        initialize=0.05,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Sales tax rate as applied to applicable capital costs",
+    )
```

Comment:
Would it make more sense to have these as flowsheet/system level variables?

Reply:
Yes I think so. How do you grab system level costing params in a unit level costing model?